rs_push

```diff
@@ -10,63 +10,88 @@
   exit 1
 fi
 
 if ! curl --version >/dev/null; then
-  echo "ERROR: cannot find curl: it is required for this script, sorry"
+  echo "ERROR: cannot find curl: it is required for this script"
   exit 1
 fi
 
 sha=`git rev-parse HEAD`
 echo "Latest commit: ${sha}"
 
-branch_name=$(git symbolic-ref -q HEAD)
-branch_name=${branch_name##refs/heads/}
-branch_name=${branch_name:-HEAD}
+branch_name=$(git rev-parse --abbrev-ref HEAD)
 echo "Current branch: ${branch_name}"
 
-echo -n "Determining 'Chef' version number for this branch ... "
-re='^([0-9][0-9]*\.[0-9][0-9]*\.)[a-z]*([0-9][0-9]*)'
-if [[ "${branch_name}" =~ $re ]]; then
-  vers="${BASH_REMATCH[1]}${BASH_REMATCH[2]}"
-  echo "using release numbering: $vers"
-else
-  vers=0.`date +%Y`.`echo "${branch_name}" | cksum | cut -d " " -f 1`
-  echo "using dev numbering: $vers"
-fi
+if [ ! $use_current_metadata ]; then
+  echo -n "Determining 'Chef' version number based on branch name..."
 
-echo "Checking 'Chef' version numbers ..."
-ok=1
-for m in `echo */metadata.rb`; do
-  vv=`egrep '^version' $m`
-  re="^version *\"${vers}\""
-  if [[ ! "$vv" =~ $re ]]; then
-    echo "  Mismatch: ${m} has ${vv}"
-    ok=
+  re='^([0-9][0-9]*\.[0-9][0-9]*\.)[a-z]*([0-9][0-9]*)'
+  if [[ "${branch_name}" =~ $re ]]; then
+    # Branch match name examples: 10.1.rc3, 10.10.2
+    # Version name from examples: 10.1.3, 10.10.2
+    vers="${BASH_REMATCH[1]}${BASH_REMATCH[2]}"
+    echo "using release numbering: $vers"
+  else
+    # Version name example: 0.2015.3015617425
+    vers=0.`date +%Y`.`echo "${branch_name}" | cksum | cut -d " " -f 1`
+    echo "using dev numbering: $vers"
   fi
-done
 
-if [[ -z "$ok" ]]; then
-  echo ""
-  echo "Please run this command to fix the version numbering ... "
-  echo sed -i -e "'"'/^version/s/"[0-9.]*"/"'${vers}'"/'"'" */metadata.rb
-  echo "Then commit the changes:"
-  echo "git commit -a -m 'Fix chef version numbers'"
-  echo "And re-run $0"
-  exit 1
+  echo "Checking 'Chef' version numbers..."
+  for metadata in */metadata.rb; do
+    current_version=`grep '^version' $metadata | awk '{print $2}' | tr -d \'\"`
+    if [[ $current_version != $version_template ]]; then
+      echo "  Mismatch: ${metadata} has ${current_version}"
+      mismatch_found=true
+    fi
+  done
+
+  if [ $mismatch_found ]; then
+    echo ""
+    echo "Please run this command to fix the version numbering ... "
+    echo ""
+    echo "  sed -i -e \"/^version/s/[\\'\\\"][0-9.]*[\\'\\\"]/'${vers}'/\" */metadata.rb"
+    echo ""
+    echo "Then commit and push the changes:"
+    echo ""
+    echo "  git commit -a -m 'Fix chef version numbers'"
+    echo "  git push"
+    echo ""
+    echo "And re-run $0"
+    exit 1
+  fi
 fi
 
-git push || exit 1
 
-# this can be fragile...
-#if [[ -e .pushrc.${branch_name} ]]; then
-#  . .pushrc.${branch_name}
-#fi
+RS_SERVER=${RS_SERVER:-us-4.rightscale.com}
 
-RS_SERVER=${RS_SERVER:-us-3.rightscale.com}
```

Comment:
Can we keep this pointed to `us-3` since that is where the RightLink account is.